code/datums/components/connect_containers.dm

```diff
@@ -0,0 +1,47 @@
+/// This component behaves similar to connect_loc_behalf, but it's nested and hooks a signal onto all MOVABLES containing this atom.
+/datum/component/connect_containers
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+	/// An assoc list of signal -> procpath to register to the loc this object is on.
+	var/list/connections
+	var/atom/movable/tracked
```

Comment:
Why do we need tracked?
(I'd ask why not just make this an element but I assume it's not hot enough. Leave a comment about that so people know not to misuse it)

Reply:
Also why do you only handle its qdeletion in `_range` and not here

Reply:
> Also why do you only handle its qdeletion in `_range` and not here

What do you mean?

Reply:
If the tracked object is deleted here it's going to hard delete, unless it's the parent of the component or the component this one is attached to. 
Regardless of that, why does the tracked var exist in the first place? I don't think you're using it for anything

Reply:
Oh wait I'm blind you're using tracked now. Deletion is still a problem

Reply:
Ditto

Reply:
Oh wait I'm blind you're using tracked now. Deletion is still a problem